### ISSUE TITLE
`dotnet new` crashes on template without `sourceName` in config

### ISSUE THREAD

**Issue by petrroll** [dotnet/templating#650]
Despite the fact that [wiki documentation](https://github.com/dotnet/templating/wiki/%22Runnable-Project%22-Templates#sourcename-optional) lists `sourceName` as an optional parameter the `dotnet new` crashes [on my templates](https://github.com/peachpiecompiler/peachpie-templates/tree/master/src/Content) when their configs don't include it.

Crash after removing `sourceName` from `peachpie-console's [config](https://github.com/peachpiecompiler/peachpie-templates/blob/master/src/Content/console-app/.template.config/template.json).
```
PS C:\Users\Petr\Downloads\tmp> dotnet new peachpie-console
Object reference not set to an instance of an object.
   at Microsoft.TemplateEngine.Orchestrator.RunnableProjects.SimpleConfigModel.ProduceOperationSetup(SpecialOperationCon
figParams defaultModel, Boolean generateMacros, ICustomFileGlobModel customGlobModel)
   at Microsoft.TemplateEngine.Orchestrator.RunnableProjects.SimpleConfigModel.Microsoft.TemplateEngine.Orchestrator.Run
nableProjects.IRunnableProjectConfig.get_OperationConfig()
   at Microsoft.TemplateEngine.Orchestrator.RunnableProjects.RunnableProjectGenerator.CreateAsync(IEngineEnvironmentSett
ings environmentSettings, ITemplate templateData, IParameterSet parameters, IComponentManager componentManager, String t
argetDirectory)
   at Microsoft.TemplateEngine.Edge.Template.TemplateCreator.<InstantiateAsync>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.TemplateEngine.Cli.New3Command.<CreateTemplateAsync>d__59.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.TemplateEngine.Cli.New3Command.<EnterSingularTemplateManipulationFlowAsync>d__64.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.TemplateEngine.Cli.New3Command.<EnterTemplateManipulationFlowAsync>d__65.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.TemplateEngine.Cli.New3Command.<ExecuteAsync>d__66.MoveNext()
```

**Comment by mlorbetske** [comment]
I believe this has been fixed for the 2.0 release. @seancpeters to confirm. The workaround is to set it to some value that is not found in your template.

**Comment by mlorbetske** [comment]
I've just verified that this has been fixed for the 2.0 release & have added a test to make sure that it doesn't regress https://github.com/dotnet/templating/pull/651
